script/v2/keeper-actions/ReallocateScUsdcV2.s.sol

```diff
@@ -0,0 +1,174 @@
+
+        console2.log("aaveV3Collateral\t", aaveV3Adapter.getCollateral(address(scUsdcV2)));
+        console2.log("aaveV3Debt\t\t", aaveV3Adapter.getDebt(address(scUsdcV2)));
+    }
+
+    function _initReallocateData() internal {
+        if (useMorpho) {
+            if (!scUsdcV2.isSupported(morphoAdapter.id())) revert("morpho adapter not supported");
+
+            _createData(morphoAdapter.id(), morphoAllocationPercent);
+
+            totalAllocationPercent += morphoAllocationPercent;
+        }
+
+        if (useAaveV2) {
+            if (!scUsdcV2.isSupported(aaveV2Adapter.id())) revert("aave v2 adapter not supported");
+
+            _createData(aaveV2Adapter.id(), aaveV2AllocationPercent);
+
+            totalAllocationPercent += aaveV2AllocationPercent;
+        }
+
+        if (useAaveV3) {
+            if (!scUsdcV2.isSupported(aaveV3Adapter.id())) revert("aave v3 adapter not supported");
+
+            _createData(aaveV3Adapter.id(), aaveV3AllocationPercent);
+
+            totalAllocationPercent += aaveV3AllocationPercent;
+        }
+
+        if (totalAllocationPercent != 1e18) {
+            revert("total allocation percent not 100%");
+        }
+    }
+
+    function _createData(uint256 _adapterId, uint256 _allocationPercent) internal {
+        ReallocateData memory data;
+
+        data.adapterId = _adapterId;
+
+        uint256 currentAllocation = scUsdcV2.getCollateral(_adapterId);
+        uint256 expectedAllocation = scUsdcV2.totalCollateral().mulWadUp(_allocationPercent);
+        uint256 currentDebt = scUsdcV2.getDebt(_adapterId);
+        uint256 expectedDebt = scUsdcV2.totalDebt().mulWadUp(_allocationPercent);
+
+        if (currentAllocation > expectedAllocation) {
+            // we need to withdraw some collateral
+            data.withdrawAmount = currentAllocation - expectedAllocation;
+            data.repayAmount = currentDebt - expectedDebt;
+        } else if (currentAllocation < expectedAllocation) {
+            // we need to supply some collateral
+            data.supplyAmount = expectedAllocation - currentAllocation;
+            data.borrowAmount = expectedDebt - currentDebt;
+        }
+
+        reallocateData.push(data);
+    }
+
+    function _createMulticallData() internal {
+        for (uint256 i = 0; i < reallocateData.length; i++) {
+            ReallocateData memory data = reallocateData[i];
+
+            if (data.withdrawAmount > 0) {
+                flashLoanAmount += data.repayAmount;
+
+                multicallData.push(abi.encodeWithSelector(scUsdcV2.repay.selector, data.adapterId, data.repayAmount));
+                multicallData.push(
+                    abi.encodeWithSelector(scUSDCv2.withdraw.selector, data.adapterId, data.withdrawAmount)
+                );
+            }
+        }
+
+        for (uint256 i = 0; i < reallocateData.length; i++) {
+            ReallocateData memory data = reallocateData[i];
+
```

Comment:
Why 2 for loops here?

Reply:
because we need to repay and withdraw in one market before we supply and borrow in another one